.github/workflows/CI.yaml

```diff
@@ -24,7 +24,21 @@ jobs:
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
+      - name: Force a fresh version of neo4j (not the cached one)
+        run: (find ~/.gradle/caches -name "*neo4j*" -exec rm -rf {} \;) || echo "All neo4j files cleaned"
       - name: Grant execute permission for gradlew
         run: chmod +x gradlew
       - name: Build with Gradle
+        env:
+          TEAMCITY_DEV_URL: ${{ secrets.TEAMCITY_DEV_URL }}
+          TEAMCITY_USER: ${{ secrets.TEAMCITY_USER }}
+          TEAMCITY_PASSWORD: ${{ secrets.TEAMCITY_PASSWORD }}
         run: ./gradlew build --continue
+      - name: Clean neo4j from the cache
+        run: (find ~/.gradle/caches -name "*neo4j*" -exec rm -rf {} \;) || echo "All neo4j files cleaned"
+      - name: Archive test results
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-results
+          path: /home/runner/work/apoc-private/apoc-private/core/build/reports/tests/test/index.html
```

Comment:
Sometimes I struggle to know what's failing in the CI exactly, cause some tests catch exceptions and still print something like FAILED or ERROR in the logs.

This step generates a zip with an html depicting what tests succeeded and which ones failed exactly:

<img width="2552" alt="" src="https://user-images.githubusercontent.com/5649971/155370338-d654619d-92b0-4b5a-9bfe-e111dc471a16.png">